src/include/network/network_state.h

```diff
@@ -22,10 +22,7 @@ enum class ConnState {
   CONN_WAIT,       // State for waiting for some event to happen
   CONN_PROCESS,    // State that runs the network protocol on received data
   CONN_CLOSING,    // State for closing the client connection
-  CONN_CLOSED,     // State for closed connection
-  CONN_INVALID,    // Invalid STate
```

Comment:
If I was right, CONN_INVALID is used to disable the system?

Reply:
It is used as a dummy value for an invalid state. It also doubles as a sensible default value since a state machine can start either on CONN_LISTENING or CONN_READ.

This does not make much sense because:
1. If an instance is not valid, it should not be there, available for interaction with the other components (object lifecycle management, an invalid object should also be out of scope)
2. It does not make much sense to have a CONN_LISTENTING state since it is disconnected with the rest of the graph. I will put in a separate PR to make that into its own callback function and take it out of the state machine.